CMakeLists.txt

```diff
@@ -121,6 +123,17 @@ endif()
 
 if(VELOX_ENABLE_PARQUET)
   add_definitions(-DVELOX_ENABLE_PARQUET)
+  # Native Parquet reader requires Apache Thrift and Arrow Parquet writer, which
+  # are included in Arrow.
+  set(VELOX_ENABLE_ARROW ON)
+endif()
+
+if(VELOX_ENABLE_ARROW)
+  set(ARROW_PREFIX "${CMAKE_CURRENT_BINARY_DIR}/third_party/arrow_ep")
```

Comment:
@yingsu00 My suggestion here is just wrapping the `thrift` that comes with Arrow in a nicer way.  No need to download it as a top level external project.  Although with my suggestion it would be easier later if we decide to bring `thrift` as a top level external project later.

Reply:
@Yuhta Jimmy thank you. I wonder if we do as https://github.com/Yuhta/velox/blob/parquet-pr/third_party/CMakeLists.txt#L35-L39, can we also build the thrift-compiler then call it in dwio/parquet/thrift/CMakeLists.txt to generate the Parquet type .cpp/.h file? In third_party/arrow_ep/src/arrow_ep-build/thrift_ep-prefix/src/thrift_ep/CMakeLists.txt, I see it depends on `BUILD_COMPILER` but don't know how to pass it in. Any ideas? 
```
if(BUILD_COMPILER)
    if(NOT EXISTS ${THRIFT_COMPILER})
        set(THRIFT_COMPILER $<TARGET_FILE:thrift-compiler>)
    endif()
    add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/compiler/cpp)
elseif(EXISTS ${THRIFT_COMPILER})
    add_executable(thrift-compiler IMPORTED)
    set_property(TARGET thrift-compiler PROPERTY IMPORTED_LOCATION ${THRIFT_COMPILER})
endif()
```

Reply:
I would say we can checked in the thrift generated code for now, this will probably simplify the internal buck build a lot.